daprdocs/content/en/java-sdk-docs/java-workflow/java-workflow-howto.md

```diff
@@ -246,4 +245,45 @@ Exiting DemoWorkflowClient.
 
 ## Next steps
 - [Learn more about Dapr workflow]({{< ref workflow-overview.md >}})
-- [Workflow API reference]({{< ref workflow_api.md >}})
\ No newline at end of file
+- [Workflow API reference]({{< ref workflow_api.md >}})
+
+## Advanced features
+
+### Task Execution Keys
+
+Task execution keys are unique identifiers generated by the durabletask-java library. They are stored in the `WorkflowActivityContext` and can be used to track and manage the execution of workflow activities. They are particularly useful for:
+
+1. **Idempotency**: Ensuring activities are not executed multiple times for the same task
+2. **State Management**: Tracking the state of activity execution
+3. **Error Handling**: Managing retries and failures in a controlled manner
+
+Here's an example of how to use task execution keys in your workflow activities:
+
+```java
+public class TaskExecutionKeyActivity implements WorkflowActivity {
+  @Override
+  public Object run(WorkflowActivityContext ctx) {
+    // Get the task execution key for this activity
+    String taskExecutionKey = ctx.getTaskExecutionKey();
+    
+    // Use the key to implement idempotency or state management
+    // For example, check if this task has already been executed
+    if (isTaskAlreadyExecuted(taskExecutionKey)) {
+      return getPreviousResult(taskExecutionKey);
+    }
+    
+    // Execute the activity logic
+    Object result = executeActivityLogic();
+    
+    // Store the result with the task execution key
+    storeResult(taskExecutionKey, result);
+    
+    return result;
+  }
+}
+```
+
```

Comment:
this is a nit, but can you remove the excess lines at the bottom. I think we only want one newline at the end of the file